queries/sun_position.py

```diff
@@ -0,0 +1,673 @@
+"""
+
+    Greynir: Natural language processing for Icelandic
+
+    Solar position query response module
+
+    Copyright (C) 2021 Miðeind ehf.
+
+       This program is free software: you can redistribute it and/or modify
+       it under the terms of the GNU General Public License as published by
+       the Free Software Foundation, either version 3 of the License, or
+       (at your option) any later version.
+       This program is distributed in the hope that it will be useful,
+       but WITHOUT ANY WARRANTY; without even the implied warranty of
+       MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+       GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see http://www.gnu.org/licenses/.
+
+
+    This module handles queries regarding time of sunrise/sunset.
+
+"""
+
+from typing import Dict, List, Iterable, Tuple, Optional, Union, cast
+
+from tree import Result, Node
+from query import Query, QueryStateDict
+
+from queries import (
+    AnswerTuple,
+    LatLonTuple,
+    MONTH_ABBREV_ORDERED,
+    is_plural,
+    gen_answer,
+)
+
+import datetime
+import logging
+import random
+import re
+import requests
+
+from bs4 import BeautifulSoup  # type: ignore
+from cachetools import TTLCache
+from settings import changedlocale
+from geo import (
+    distance,
+    in_iceland,
+    capitalize_placename,
+    ICE_PLACENAME_BLACKLIST,
+)
+from iceaddr import placename_lookup  # type: ignore
+
```

Comment:
Perhaps add TODO at top of file:  "Hvenær rís sólin [dagsetning]?"

Reply:
Added.